### PR TITLE
Pull request listing.

### PR DESCRIPTION
**Pull request by seanparsons**
**Problem:**
Currently there is no support for handling pull requests related to the branch.

**Fix:**
Along with a link to create a new PR, existing PRs for the current branch are listed which can be clicked on to access them.

**Screenshots:**
image

**Commit Details:**
- Broke apart `GithubOperation` into separate types.
- Added new `GithubOperation` case of `GithubListPullRequestsForBranch`.
- Added `currentBranchPullRequests` to `GithubData`.
- Added new React component `PullRequestPane` which contains the listing of pull requests.
- Added `updatePullRequestsForBranch` and included a call to it in `refreshGithubData`.
- Slightly tweaked the endpoint URL for loading the content of a branch to make it more consistent with the new endpoint for pulling the PRs for a branch.
- Added endpoint to get the pull requests for a branch.
- Implemented `listPullRequests` to retrieve the pull requests for a branch.

### CODE REVIEW

File: editor/src/components/editor/store/editor-state.ts [hunk 1]
```diff
@@ -255,12 +255,41 @@ export interface UserState extends UserConfiguration {
...
-  | { name: 'listBranches' }
-  | { name: 'loadBranch'; branchName: string; githubRepo: GithubRepo }
-  | { name: 'loadRepositories' }
-  | { name: 'updateAgainstBranch' }
+  | GithubCommish
+  | GithubListBranches
+  | GithubLoadBranch
+  | GithubLoadRepositories
+  | GithubUpdateAgainstBranch
+  | GithubListPullRequestsForBranch
```
Comment (ruggi) [inline editor/src/components/editor/store/editor-state.ts:292]: very nice split 👌

File: editor/src/core/shared/github.ts [hunk 2]
```diff
@@ -406,7 +482,7 @@ export async function updateProjectWithBranchContent(
         dispatch(
           [
             updateGithubChecksums(getProjectContentsChecksums(responseBody.content)),
-            updateProjectContents(responseBody.content),
+            //updateProjectContents(responseBody.content),
```
Comment (ruggi) [inline editor/src/core/shared/github.ts:485]: 🧐

Comment (seanparsons) [reply editor/src/core/shared/github.ts:485]: Gah, I did mean to undo that!

Comment (seanparsons) [reply editor/src/core/shared/github.ts]: Fixed in 20d22ccb9.

File: editor/src/components/editor/store/editor-state.ts [hunk 3]
```diff
@@ -315,6 +347,21 @@ export function isGithubUpdating(operations: Array<GithubOperation>): boolean {
   return operations.some((operation) => operation.name === 'updateAgainstBranch')
 }
 
+export function isGithubListingPullRequestsForBranch(
+  operations: Array<GithubOperation>,
+
```
Comment (Rheeseyb) [inline editor/src/components/editor/store/editor-state.ts:352]:
```suggestion
```
Strange whitespace here

File: editor/src/components/navigator/left-pane/github-pane/pull-request-pane.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        ...(pullRequests ?? []),
+        {
+          title: 'Create PR on Github.',
+          htmlURL: createPullRequestURLWithParams,
+        },
+      ]
+    } else {
+      return null
+    }
+  }, [pullRequests, githubRepo, branchName])
```
Comment (Rheeseyb) [inline editor/src/components/navigator/left-pane/github-pane/pull-request-pane.tsx:72]: This is very cool, but I suspect we'll likely need to trigger a commit+push as part of this (or prompt the user to if there are any unsaved changes). Probably more a thought for follow up work.

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/components/editor/store/editor-state.ts (hunk 3 above)
Comment (seanparsons) [reply editor/src/components/editor/store/editor-state.ts:352]: I'm surprised that didn't get obliterated by Prettier.

**Pull request merged**